salt/states/zabbix_host.py

```diff
@@ -397,12 +415,19 @@ def _interface_format(interfaces_data):
                 # combine connection_args, inventory, and clear_old
                 sum_kwargs = dict(new_inventory)
                 sum_kwargs.update(connection_args)
-                sum_kwargs["clear_old"] = True
+                sum_kwargs["clear_old"] = inventory_clean
 
                 hostupdate = __salt__["zabbix.host_inventory_set"](hostid, **sum_kwargs)
                 ret["changes"]["inventory"] = str(new_inventory)
                 if "error" in hostupdate:
                     error.append(hostupdate["error"])
+                # restore inventory_mode as host_inventory_set changes it to '0'
```

Comment:
Can `host_inventory_set` not be changed to take an `inventory_mode`?

Reply:
Yes, it could, but I guess it's a state related statement and the state module is a better place for it.

If you are thinking about avoid send this additional `host_update`, it can´t be done as the `inventory_mode` needs to be set to `0` to being able to set the inventory. If the `inventory_mode` is `1` and you try to manually set the inventory you'll get a permission error:

`Zabbix API: Invalid params. (No permissions to referred object or it does not exist!)`

So, as there is a need of that additional `host_update` in someplace and its use is to preserve the `inventory_mode` state, I thought the code should be in the state module instead in the execution module.

Reply:
It might depend on the version of Zabbix. Older releases had a lot of bugs with inventory in the API.

Reply:
> It might depend on the version of Zabbix. Older releases had a lot of bugs with inventory in the API.

Maybe, but my test error was due lack of attention. I removed and recreated the host but send the tests with the old ID. Of course the `referred object ... does not exist!`.

Reply:
Done